micronaut-build-plugin-sourcegen-generator/src/test/groovy/io/micronaut/sourcegen/generator/visitors/GradlePluginGenerationSpec.groovy

```diff
@@ -0,0 +1,209 @@
+ * Specification that is used for configuring Wolf task.
+ * Wolf Gradle task.
+ */
+public interface WolfSpec {
+  /**
+   * Configurable slogan parameter.
+   */
+  Property<String> getSlogan();
+
+  /**
+   * Configurable age parameter.
+   */
+  Property<Integer> getAge();
+}"""
+
+        var extensionContent = stripImports(files.get("test.WolfExtension").getCharContent(false))
+        extensionContent == """/**
+ * Configures the Wolf execution.
+ */
+public interface WolfExtension {
+  /**
+   * Create a task for awooo.
+   * Main execution of Wolf task.
+   * @param name The unique identifier used to derive task names
+   * @param spec The configurable specification
+   */
+  void awooo(String name, Action<WolfSpec> action);
+}"""
+
+        var defaultExtensionContent = stripImports(files.get("test.DefaultWolfExtension").getCharContent(false))
+        defaultExtensionContent == """public abstract class DefaultWolfExtension implements WolfExtension {
+  protected final Set<String> names = new java.util.HashSet();
+
+  protected final Project project;
+
+  protected final Configuration classpath;
+
+  @Inject
+  public DefaultWolfExtension(Project project, Configuration classpath) {
+    this.project = project;
+    this.classpath = classpath;
+  }
+
+  public void awooo(String name, Action<WolfSpec> action) {
+    if (!this.names.add(name)) {
+      throw new org.gradle.api.GradleException(String.format("An awooo definition with name '%s' was already created", name));
+    }
+    WolfSpec spec = this.project.getObjects().newInstance(WolfSpec.class);
+    this.configureSpec(spec);
+    action.execute(spec);
+    TaskProvider<? extends WolfTask> task = this.createWolfTask(name, new WolfTaskConfigurator(spec, this.classpath));
+  }
+
+  TaskProvider<? extends WolfTask> createWolfTask(String name, Action<WolfTask> configurator) {
+    return this.project.getTasks().register(name, WolfTask.class, configurator);
+  }
+
+  protected void configureSpec(WolfSpec spec) {
+    spec.getAge().convention(1);
+  }
+
+  protected static class WolfTaskConfigurator implements Action<WolfTask> {
+    WolfSpec spec;
+
+    Configuration classpath;
+
+    WolfTaskConfigurator(WolfSpec spec, Configuration classpath) {
+      this.spec = spec;
+      this.classpath = classpath;
+    }
+
+    public void execute(WolfTask arg1) {
+      arg1.getClasspath().from(this.classpath);
+      arg1.setDescription("Configure the awooo");
+      arg1.getSlogan().convention(this.spec.getSlogan());
+      arg1.getAge().convention(this.spec.getAge());
+    }
+  }
+}"""
+
+        var pluginContent = stripImports(files.get("test.WolfPlugin").getCharContent(false))
+        pluginContent == """public class WolfPlugin implements Plugin<Project> {
+  protected WolfExtension createExtension(Project project, Configuration classpath) {
+    return project.getExtensions().create(WolfExtension.class, "Wolf", DefaultWolfExtension.class, project, classpath);
```

Comment:
The defaults should probably be handled at the extension level, rather than in the task execution.

Reply:
They are handled on both levels. Not sure if required, but that seems to be the way for current plugins. I suppose this allows using task on its own.

Reply:
mmm, usually the tasks default to (`convention()`) the extension, which itself has `convention()`. That should be good.